Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -0,0 +1,316 @@
+        return blocklists.reduce(TimeInterval(0)) { result, filename in
+            guard let path = Bundle.main.path(forResource: filename, ofType: "json") else { return result }
+            let date = lastModifiedSince1970(path: path) ?? 0
+            return date > result ? date : result
+        }
+    }
+
+    fileprivate func removeAllRulesInStore(completion: @escaping () -> Void) {
+        guard let ruleStore = ruleStore else { return }
+
+        ruleStore.getAvailableContentRuleListIdentifiers { available in
+            guard let available = available else {
+                completion()
+                return
+            }
+            var completionCount = 0
+            available.forEach {
+                ruleStore.removeContentRuleList(forIdentifier: $0) { _ in
+                    completionCount += 1
+                    if completionCount == available.count {
+                        completion()
+                    }
+                }
+            }
+        }
+    }
+
+    // If any blocker files are newer than the date saved in prefs,
+    // remove all the content blockers and reload them.
+    // Pass true to completion block if the rule store was modified.
+    fileprivate func removeOldListsByDateFromStore(completion: @escaping (_ modified: Bool) -> Void) {
+        let fileDate = self.dateOfMostRecentBlockerFile()
+        let prefsNewestDate = UserDefaults.standard.double(forKey: "blocker-file-date")
+        if prefsNewestDate < 1 || fileDate <= prefsNewestDate {
+            completion(false)
+            return
+        }
+
+        UserDefaults.standard.set(fileDate, forKey: "blocker-file-date")
+        self.removeAllRulesInStore() {
+            completion(true)
+        }
+    }
+
+    // Pass true to completion block if the rule store was modified.
+    fileprivate func removeOldListsByNameFromStore(completion: @escaping (_ modified: Bool) -> Void) {
+        guard let ruleStore = ruleStore else { return }
+        var noMatchingIdentifierFoundForRule = false
+
+        ruleStore.getAvailableContentRuleListIdentifiers { available in
+            guard let available = available else {
+                completion(false)
+                return
+            }
+
+            let blocklists = self.blocklistBasic + self.blocklistStrict
+            for contentRuleIdentifier in available {
+                if !blocklists.contains(where: { $0 == contentRuleIdentifier }) {
+                    noMatchingIdentifierFoundForRule = true
+                    break
+                }
+            }
+
+            let fileDate = self.dateOfMostRecentBlockerFile()
+            let prefsNewestDate = UserDefaults.standard.double(forKey: "blocker-file-date")
+            if fileDate <= prefsNewestDate && !noMatchingIdentifierFoundForRule {
+                completion(false)
+                return
+            }
+            UserDefaults.standard.set(fileDate, forKey: "blocker-file-date")
```

Comment:
Why are you setting userdefaults directly here? Shouldnt these be set via prefs?